### PR TITLE
tests: re-execute the failed tests when "Run failed" label is set in the PR 

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change is done to save the failed tests for a run and execute those
tests when the labed "Run failed" is set in the PR.

The idea of this is to speed up the re-executions

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -292,47 +292,82 @@ jobs:
...
+          # Save previous failed test results in FAILED_TESTS env var
+          FAILED_TESTS=""
+          if [ -f "$FAILED_TESTS_FILE" ]; then
+              echo "Failed tests file found"
+              FAILED_TESTS="$(cat $FAILED_TESTS_FILE)"
+          fi
+          if [ -n "$FAILED_TESTS" ]; then
+              echo "Failed tests to run: $FAILED_TESTS"
+              echo "FAILED_TESTS=$FAILED_TESTS" >> $GITHUB_ENV
+          fi
```
Comment (mardy) [inline .github/workflows/test.yaml:337]: nitpick: this could be nested in the `if` above

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -350,6 +385,28 @@ jobs:
...
+          if [ -f spread.log ]; then
+              echo "Running spread log analyzer"
+
+              # Aborted cannot be used to determine if we need to re-run all because it could be a aborted an entire suite
+              if grep -q "Aborted tasks: 0" spread.log; then
+                  ./tests/lib/external/snapd-testing-tools/utils/log-parser spread.log --output spread-results.json
+                  FAILED_TESTS="$(jq -r '.[] | select( .type == "result") | select( .result_type == "Failed") | select( .level == "task" or .level == "tasks" ) | .detail.lines[]' spread-results.json | cut -d '-' -f2- | xargs)"
+
+                  echo "Failed tests saved for re-execution: $FAILED_TESTS"
+                  echo "$FAILED_TESTS" > "$FAILED_TESTS_FILE"
```
Comment (mardy) [inline .github/workflows/test.yaml:400]: Are we sure that we won't hit a limit on command line length (or number of parameters) when calling `xargs`? I've seen that happening sometimes when running `find | xargs ...`, but I don't know what the limit is.
Can't we just do
```
jq -r '.[] | select( .type == "result") | select( .result_type == "Failed") | select( .level == "task" or .level == "tasks" ) | .detail.lines[]' spread-results.json \
    | cut -d '-' -f2- > "$FAILED_TEST_FILE"
```
?

**Review by mardy — APPROVED** [review]
LGTM, with a question on `xargs`

File: .github/workflows/test.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply .github/workflows/test.yaml:337]: yes, updated, thanks

File: .github/workflows/test.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply .github/workflows/test.yaml:400]: agree, updated, thanks for the review

**Review by MiguelPires — APPROVED** [review]
LGTM

**Pull request merged**